sharding/xorValidatorShuffler.go

```diff
@@ -73,9 +71,9 @@ func (rxs *randXORShuffler) UpdateParams(
 //          b)  In case (shuffled out nodes + new nodes) < (nbShards * perShardHysteresis) then we can immediately
 //              execute the shard merge
 //          c)  No change in the number of shards then nothing extra needs to be done
-func (rxs *randXORShuffler) UpdateNodeLists(args ArgsUpdateNodes) (map[uint32][]Validator, map[uint32][]Validator, []Validator) {
-	eligibleAfterReshard := copyValidatorMap(args.Eligible)
-	waitingAfterReshard := copyValidatorMap(args.Waiting)
+func (rxs *randXORShuffler) UpdateNodeLists(args ArgsUpdateNodes) ([][]Validator, [][]Validator, []Validator) {
```

Comment:
I think receiving as map and returning as map should remain, as it is cleaner for the component which calls on the top. Doing the translation from map to slice could happen here in this public method, and all the other method could remain with lists of list.

Reply:
undone and used sorted keys on map